### PR TITLE
books

### PR DESCRIPTION
**Pull request by Gavrilik**

### CODE REVIEW

File: src/features/book/book.json-repository.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    return book;
+  }
+  update(id: number, updateBookDto: UpdateBookDto) {
+    books = books.map((book) => {
+      if (book.id === id) {
+        return { ...book, ...updateBookDto };
+      }
+      return this.findOne(id);
+    });
+   // throw new NotFoundException(this.findOne(id));
```
Comment (dmialeshka) [inline src/features/book/book.json-repository.ts:40]: remove

File: src/features/book/book.json-repository.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  }
+  remove(id: number) {
+    this.findOne(id);
+    const condition = (book) => book.id !== id;
+    const filteredBooks = books.filter(condition);
+    return (books = filteredBooks);
```
Comment (dmialeshka) [inline src/features/book/book.json-repository.ts:46]: Please, explain what do you do here

File: src/features/book/book.module.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import { Module } from '@nestjs/common';
 import { BookService } from './book.service';
 import { BookController } from './book.controller';
-import { UserModule } from '../user/user.module';
+//import { UserModule } from '../user/user.module';
```
Comment (dmialeshka) [inline src/features/book/book.module.ts:4]: unused import

File: src/features/book/book.module.ts [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
+import { BookJsonRepository } from './book.json-repository';
 
 @Module({
   controllers: [BookController],
-  providers: [BookService],
-  imports: [UserModule],
+  providers: [BookService, BookJsonRepository],
+  exports: [BookService],
```
Comment (dmialeshka) [inline src/features/book/book.module.ts:10]: Do we need use bookService in different module ?

**Review by dmialeshka — CHANGES_REQUESTED** [review]

File: src/features/book/book.json-repository.ts (hunk 1 above)
Comment (Gavrilik) [reply src/features/book/book.json-repository.ts]: delete

File: src/features/book/book.json-repository.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+  findOne(id: number) {
+    const condition = (b: CreateBookDto) => b.id === id;
+    const book = books.find(condition);
+    if (!book) {
+      throw new NotFoundException(`book not found!, ${id}`);
+    }
+    return book;
+  }
+  update(id: number, updateBookDto: UpdateBookDto) {
```
Comment (dmialeshka) [inline src/features/book/book.json-repository.ts:33]: fix format

File: src/features/book/book.module.ts (hunk 3 above)
Comment (Gavrilik) [reply src/features/book/book.module.ts:5]: delete

File: src/features/book/book.module.ts (hunk 4 above)
Comment (Gavrilik) [reply src/features/book/book.module.ts:11]: fixed

File: src/features/book/book.module.ts (hunk 3 above)
Comment (Gavrilik) [reply src/features/book/book.module.ts:5]: deleted

File: src/features/book/book.json-repository.ts (hunk 2 above)
Comment (Gavrilik) [reply src/features/book/book.json-repository.ts:45]: 1. check for book availability
2. arrow function with a condition for the filter operation
3. call the filter operation for 'books' with a reference to the condition
4. assignment operation

File: src/features/book/book.json-repository.ts (hunk 5 above)
Comment (Gavrilik) [reply src/features/book/book.json-repository.ts:33]: fixed

File: src/features/book/book.json-repository.ts (hunk 2 above)
Comment (dmialeshka) [reply src/features/book/book.json-repository.ts:48]: Do you think the assignment operation is good as returning a result?

File: src/features/book/book.module.ts (hunk 4 above)
Comment (dmialeshka) [reply src/features/book/book.module.ts:10]: What are you fixing here?

File: src/features/book/book.service.ts [hunk 6]
```diff
@@ -1,28 +1,31 @@
+/* eslint-disable prettier/prettier */
 import { Injectable } from '@nestjs/common';
+import { BookJsonRepository } from './book.json-repository';
 import { CreateBookDto } from './dto/create-book.dto';
 import { UpdateBookDto } from './dto/update-book.dto';
 
 @Injectable()
 export class BookService {
+  constructor(private readonly bookJsonRepository: BookJsonRepository) {}
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (dmialeshka) [inline src/features/book/book.service.ts:10]: remove

File: src/features/book/book.service.ts [hunk 7]
```diff
@@ -1,28 +1,31 @@
...
-    return `This action returns all books`;
+    return this.bookJsonRepository.findAll();
   }
 
   findOne(id: number) {
-    return `This action returns a #${id} book`;
+    return this.bookJsonRepository.findOne(id);
   }
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (dmialeshka) [inline src/features/book/book.service.ts:23]: remove

File: src/features/book/dto/create-book.dto.ts [hunk 8]
```diff
@@ -1 +1,13 @@
-export class CreateBookDto {}
+/* eslint-disable prettier/prettier */
```
Comment (dmialeshka) [inline src/features/book/dto/create-book.dto.ts:1]: Why we disable prettier here ?

**Review by dmialeshka — CHANGES_REQUESTED** [review]

File: src/features/book/book.json-repository.ts (hunk 2 above)
Comment (Gavrilik) [reply src/features/book/book.json-repository.ts:48]: yes, it returns the assigned value.

File: src/features/book/book.module.ts (hunk 4 above)
Comment (Gavrilik) [reply src/features/book/book.module.ts:10]: add bookJsonRepository in providers and export Bookservice.
deleted unused imports.

File: src/features/book/book.json-repository.ts (hunk 2 above)
Comment (dmialeshka) [reply src/features/book/book.json-repository.ts:48]: Please do not do 2 things at the same time,
1. do assignment operation
2. return value

File: src/features/book/book.module.ts (hunk 4 above)
Comment (dmialeshka) [reply src/features/book/book.module.ts:10]: Do we really need use BookService outside the Book Module ?

File: src/features/book/dto/create-book.dto.ts (hunk 8 above)
Comment (Gavrilik) [reply src/features/book/dto/create-book.dto.ts:1]: To exclude files from formatting.
To make the code look better

Comment (dmialeshka) [reply src/features/book/dto/create-book.dto.ts:1]: Everything that you described above you disabled with this line :)

**Review by dmialeshka — APPROVED** [review]

**Pull request merged**